vc-got.el

```diff
@@ -460,6 +470,22 @@ files on disk."
 ;;;###autoload     (load "vc-got" nil t)
 ;;;###autoload     (vc-got-registered file)))
 
+(defun vc-got-create-repo ()
+  "Creates an empty repository with `got init' in the current directory and
+populates it with files from a directory polled from user."
+  (let ((repo-directory (expand-file-name default-directory)))
+    (unless (directory-empty-p repo-directory)
+      (error "The new repository directory must be empty"))
+    (let ((import-directory (expand-file-name
+                             (read-directory-name "What directory to import from: "))))
+      (when (string= repo-directory import-directory)
+        (error "Cannot import the repo directory"))
+      (apply #'vc-got-command nil 0 repo-directory "init"
+             (ensure-list vc-got-create-repo-init-switches))
+      (apply #'vc-got-command t 'async import-directory "import"
+             (append (list "-m" (read-string "Message for import commit: "))
+                     (ensure-list vc-got-create-repo-import-switches))))))
```

Comment:
nit: I think this should also run `got checkout -E /path/to/repo /current/directory/`, so that we can track the files we've imported.

maybe it's a bit opinionated, but I do that all the times and I think it's fair to expect this behaviour